phase2/kubeadm/configure-vm-kubeadm-master.sh

```diff
@@ -15,24 +15,67 @@ elif [[ "$KUBEADM_CNI_PLUGIN" == "weave" ]]; then
 fi
 
 mkdir -p $KUBEADM_DIR
+
+# break down the version string
+KUBEADM_KUBERNETES_VERSION_MAJOR=`cut -d'.' -f 1 <<< $KUBEADM_KUBERNETES_VERSION | cut -d'v' -f 2`
+KUBEADM_KUBERNETES_VERSION_MINOR=`cut -d'.' -f 2 <<< $KUBEADM_KUBERNETES_VERSION`
+KUBEADM_KUBERNETES_VERSION_PATCH=`cut -d'.' -f 3 <<< $KUBEADM_KUBERNETES_VERSION`
+
+# set defaults
 cat <<EOF |tee $KUBEADM_CONFIG_FILE
 kind: MasterConfiguration
-apiVersion: kubeadm.k8s.io/v1alpha1
+kubernetesVersion: "$KUBEADM_KUBERNETES_VERSION"
 api:
   advertiseAddress: "$KUBEADM_ADVERTISE_ADDRESSES"
   bindPort: 443
 networking:
   podSubnet: "$POD_NETWORK_CIDR"
-kubernetesVersion: "$KUBEADM_KUBERNETES_VERSION"
-token: "$KUBEADM_TOKEN"
 EOF
 
-if [[ "$KUBEADM_ENABLE_CLOUD_PROVIDER" == true ]]; then
+# handle v1alpha1
+########################################################################
+if [ "$KUBEADM_KUBERNETES_VERSION_MINOR" -lt "11" ]; then
```

Comment:
nit: the rest of this script looks to be bash leveraging `[[`, why not here for consistency?